src/vector/_compute/spatial/eta.py

```diff
@@ -30,13 +30,11 @@
 )
 
 
+# TODO: revert back to `nan_to_num` implementation once
+# https://github.com/cupy/cupy/issues/9143 is fixed
+# `lib.where` works but there is no SymPy equivalent for the function
 def xy_z(lib, x, y, z):
-    return lib.nan_to_num(
-        lib.arcsinh(z / lib.sqrt(x**2 + y**2)),
-        nan=lib.nan_to_num((z != 0) * inf, posinf=nan),
-        posinf=inf,
-        neginf=-inf,
-    )
+    return lib.where(z != 0, lib.arcsinh(z / lib.sqrt(x**2 + y**2)), z) * 1
```

Comment:
There is absolutely no issue in switching to this definition for the numerical backends, so this is not a hack.

Reply:
Thanks, @pfackeldey! TIL.

I have updated the PR with nested `where`s.

(We do really need GPU/Dask/JAX integration tests for Scikit-HEP...)